src/ui/ImageReference.hh

```diff
@@ -116,21 +119,30 @@ public:
             pRepository.ReleaseReference(*this);
         }
     }
-    
+
+private:
+    unsigned long GetData() noexcept { return GetDataImpl(*this); }
     /// <summary>
     /// Gets custom data associated to the reference - used to cache data by <see cref="ISurface::DrawImage" />.
     /// </summary>
-    unsigned long GetData() const noexcept { return m_nData; }
+    const unsigned long GetData() const noexcept { return GetDataImpl(*this); }
+    
+    template<typename T>
+    static auto GetDataImpl(T& ir) noexcept -> decltype(ir.GetData())
+    {
+        return ir.m_nData;
+    }
 
     /// <summary>
     /// Sets custom data associated to the reference - used to cache data by <see cref="ISurface::DrawImage" />.
     /// </summary>
-    void SetData(unsigned long nValue) noexcept { m_nData = nValue; }
+    inline constexpr void SetData(unsigned long nValue) noexcept { m_nData = nValue; }
 
-private:
-    ImageType m_nType = ImageType::None;
+    ImageType m_nType{};
     std::string m_sName;
-    unsigned long m_nData{};
+    mutable unsigned long m_nData{};
+    friend class drawing::gdi::ImageRepository;
+    friend class drawing::gdi::GDISurface;
```

Comment:
This shouldn't be necessary. Only the helper function should be directly interacting with the `m_nData` member.

Reply:
I thought you wanted Get/SetData as private?

Reply:
It won't compile otherwise.